couchdbkit/client.py

```diff
@@ -175,7 +176,10 @@ def delete_db(self, dbname):
         """
         Delete database
         """
-        del self[dbname]
+        try:
+            del self[dbname]
+        except CloudantClientException:
+            raise ResourceNotFound
```

Comment:
Should we add a message here from the exception that was given? Or maybe just `ResourceNotFound('Database x doesn't exist and cannot be deleted')` ? Although its possible that deleting a non existent database could just return silently

Reply:
Updated to preserve the error message provided by cloudant.  Example:

"Database test_commcarehq__fluff-bihar does not exist. Verify that the client is valid and try again."